### PR TITLE
initial specs

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

This PR consists of initial specifications, including details about the generic DA interface implementation for Celestia backend.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by distractedm1nd** [comment]
incredibly based

File: specs/src/celestia-da.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+### Commit
+
+Commit returns the commitment to blobs.
+
+The implementation calls `blob.CreateCommitments` which does not call any RPC method, so it's completely offline.
+
+### Submit
+
+Submit submits blobs and returns their ids and proofs.
```
Comment (nashqueue) [inline specs/src/celestia-da.md:53]: Could you talk about how you can use options? We might want to use that in the interface.

File: specs/src/celestia-da.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+## Assumptions
+
+There should be a local celestia node, either full, bridge or light node running and accessible from the implementation.
+
+The implementation uses the celestia node to interact with the celestia network to send, receive and validate blobs.
+
+To be able to submit blobs, the following assumptions are made:
+
+* The local celestia node is fully caught up with the network to.
```
Comment (MSevey) [inline specs/src/celestia-da.md:32]: network tip?

File: specs/src/celestia-da.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Validate validates blob ids and proofs and returns whether they are included.
+
+The implementation calls [blob.Included](https://docs.celestia.org/api/v0.11.0-rc13/#blob.Included) RPC method on the Celestia Node API.
+
+## References
+[1] https://github.com/rollkit/go-da
+
+[2] https://github.com/rollkit/celestia-openrpc
+
+[3] https://docs.celestia.org/api/v0.11.0-rc13/
```
Comment (MSevey) [inline specs/src/celestia-da.md:73]: You can further refactor the links like this
```suggestion
[3] [node api docs]

[node api docs]: https://docs.celestia.org/api/v0.11.0-rc13/
```

then you can use that same link in the rest of the readme so you only have the update the API url once when the version bumps.

Comment (MSevey) [reply specs/src/celestia-da.md:73]: same can be applied to the go-da and celestia-openrpc links

File: specs/src/celestia-da.md (hunk 2 above)
Comment (tuxcanfly) [reply specs/src/celestia-da.md:32]: Fixed.

File: specs/src/celestia-da.md (hunk 3 above)
Comment (tuxcanfly) [reply specs/src/celestia-da.md:73]: Thanks, updated.

File: specs/src/celestia-da.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+### Submit
+
+Submit submits blobs and returns their ids and proofs.
+
+The implementation calls [blob.Submit](https://docs.celestia.org/api/v0.11.0-rc13/#blob.Submit) RPC method with `DefaultSubmitOptions` on the Celestia Node API.
+
+`DefaultSubmitOptions` uses default values for `Fee` and `GasLimit`.
+
+TODO: Allow `SubmitOptions` to be passed in to the Submit method.
```
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply specs/src/celestia-da.md:60]: OK, this is related to go-da, so I'll create an issue there.

File: specs/src/celestia-da.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+## Abstract
+
+This package implements the generic DA interface defined in [go-da] for Celestia.
+
+## Details
+
+The generic DA interface defines how DA implementations can submit, retrieve and validate blobs.
+
+The Celestia implementation is a wrapper around [celestia-openrpc] which
```
Comment (nashqueue) [inline specs/src/celestia-da.md:11]: Does not wrap celestia openrpc so this has to be adapted

File: specs/src/celestia-da.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+There should be a local celestia node, either full, bridge or light node running and accessible from the implementation.
+
+The implementation uses the celestia node to interact with the celestia network to send, receive and validate blobs.
+
+To be able to submit blobs, the following assumptions are made:
+
+* The local celestia node is fully caught up with the network tip.
+* The auth token has at least read/write permissions.
+* The local celestia node has a account with a non zero balance to pay fee.
```
Comment (nashqueue) [inline specs/src/celestia-da.md:34]:
```suggestion
* The local celestia node has an account with a non-zero balance to pay fees if it wants to send blobs or transactions. You don`t need a balance for requests 
```

**Review by nashqueue — CHANGES_REQUESTED** [review]

File: specs/src/celestia-da.md (hunk 5 above)
Comment (tuxcanfly) [reply specs/src/celestia-da.md]: Done.

File: specs/src/celestia-da.md (hunk 6 above)
Comment (tuxcanfly) [reply specs/src/celestia-da.md]: OK. Removed 'You' for consistency.

**Comment by tuxcanfly** [comment]
Updated to mention `GetIDs` in the spec.

**Review by nashqueue — APPROVED** [review]

**Review by MSevey — APPROVED** [review]

**Pull request merged**